roles/varnish/templates/etc/varnish/varnish.vcl

```diff
@@ -302,301 +295,142 @@ sub vcl_recv {
-
-        #     lastmod = VRT_GetHdr(sp, HDR_BERESP, "\016Last-Modified:");
-        #     if (lastmod) {
-        #         lmod =  TIM_parse(lastmod);
-        #         age = TIM_real() - lmod;
-        #         VRT_l_beresp_ttl(sp, age*factor);
-        #     }
-        #  }C
-
-        # This is the attempted replacement, but it fails to compile.
-        # set beresp.ttl = std.time(beresp.http.last-modified, now);
-        # /FIXME
-        set beresp.http.X-FACTOR-TTL = "ttl: " + beresp.ttl;
-    }
+    # Expose the backend and ttl as diagnostic information
+    set beresp.http.X-Varnish-Backend = beresp.backend.name;
+    set beresp.http.X-Varnish-Ttl = beresp.ttl;
 
-    if (bereq.url ~ "content/[^/]*/[0-9.]*/(\?format=)?pdf$") {
-        set beresp.http.X-My-Header = "VersionedPDF";
-        set beresp.uncacheable = true;
-        return(deliver);
-    }
-    if (bereq.url ~ "content/[^/]*/latest/(\?format=)?pdf$") {
-        set beresp.http.X-My-Header = "LatestPDF";
-        set beresp.uncacheable = true;
-        return(deliver);
-    }
-    if (bereq.url ~ "content/[^/]*/[0-9.]*/offline$") {
-        set beresp.http.X-My-Header = "VersionedOfflineZip";
-        set beresp.uncacheable = true;
-        return(deliver);
-    }
-    if (bereq.url ~ "content/[^/]*/[0-9.]*/complete$") {
-        set beresp.http.X-My-Header = "VersionedCompleteZip";
-        set beresp.uncacheable = true;
-        return(deliver);
-    }
-    call rewrite_s_maxage;
-    set beresp.http.X-FACTOR-TTL = "ttl: " + beresp.ttl;
-    return(deliver);
+    # Avoid the builtin vcl_backend_response, we already handled everything ourselves
+    return (deliver);
 }
 
-sub vcl_backend_error {
-    if (beresp.status == 750) {
-        set beresp.http.Location = "http://" + regsub(bereq.http.host, "^[^:]*", "{{ frontend_domain }}") + bereq.url;
-        set beresp.status = 301;
-        return(deliver);
-    } elsif (beresp.status == 700) {
-        set beresp.http.Location = bereq.http.host + regsub(bereq.url, "//$", "/");
-        set beresp.status = 301;
-        return(deliver);
+# https://varnish-cache.org/tips/vcl/redirect.html
+# Allows the use of synth(301, url) and synth(302, url)
+sub vcl_synth {
+    if (resp.status == 301 || resp.status == 302) {
+        set resp.http.location = resp.reason;
+        set resp.reason = "Moved";
+        return (deliver);
     }
-}
 
-sub vcl_deliver {
-        if (obj.hits > 0) {
-                set resp.http.X-Cache = "HIT";
-        } else {
-                set resp.http.X-Cache = "MISS";
-        }
-    call rewrite_age;
+    # Run builtin vcl_synth
 }
 
 sub vcl_hash {
-    hash_data(req.url);
-    if (req.http.host) {
-        hash_data(req.http.host);
-    } else {
-        hash_data(server.ip);
-    }
-    if (req.http.Accept ~ "application/xhtml\+xml" && req.url ~ "^/contents/") {
+    # The response to this archive route varies depending on the Accept header
+    # This works like a normalized Vary: Accept
```

Comment:
Right, the previous hash of host was because this varnish would receive reqs for 'cnx.org', 'legacy.cnx.org' and even 'archive.cnx.org' - I _think_ we've got that sorted in the higher-order routing now - Hmm, don't we still get 'cnx.org' and 'archive' urls here? We serve very different results … i.e. cnx.org/search?q=string returns the standard webview app page, archive.cnx.org/search?q=string returns the json search results - pretty sure this will return the webview html for a search, as written. (also true for /contents, etc.)

Reply:
Covered by the builtin vcl_hash (it hashes the host or the ip, if no host).

Reply:
ah, good enough.